### PR TITLE
Implements DebugDisplay classes.

### PR DESCRIPTION
**Pull request by OmniBlade**

### CODE REVIEW

File: src/game/client/w3ddebugdisplay.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3ddebugdisplay.h"
+#include "displaystringmanager.h"
+#include "displaystring.h"
+
+// temp location
+int GameMakeColor(unsigned char red, unsigned char green, unsigned char blue, unsigned char alpha)
```
Comment (tomsons26) [inline src/game/client/w3ddebugdisplay.cpp]: a home for this needs to be figured out

File: src/game/client/w3ddebugdisplay.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+void W3DDebugDisplay::Draw_Text(int x, int y, char *text)
+{
+    if (m_font != nullptr && m_displayString != nullptr) {
+        Utf16String unicode;
+
+        unicode.Translate(Utf8String(text));
+
+        m_displayString->Set_Text(unicode);
+        m_displayString->Draw(
```
Comment (tomsons26) [inline src/game/client/w3ddebugdisplay.cpp]: wonky formatting?

File: src/game/client/debugdisplay.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ */
+#include "debugdisplay.h"
+#include <stdio.h>
+#include <stdarg.h>
+
+void DebugDisplay::Printf(char *format, ...)
+{
+    static char _text[5120];
+
+    va_list va;
```
Comment (tomsons26) [inline src/game/client/debugdisplay.cpp]: couldn't figure out how to clean this function up

File: src/game/client/w3ddebugdisplay.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    void Set_Font(GameFont *font);
+    void Set_Font_Width(int width) { m_fontWidth = width; }
+    void Set_Font_Height(int height) { m_fontHeight = height; }
+
+private:
+    GameFont *m_font;
+    int m_fontWidth;
+    int m_fontHeight;
+    DisplayString *m_displayString;
+};
```
Comment (tomsons26) [inline src/game/client/w3ddebugdisplay.h]: missing new line

File: src/game/client/debugdisplay.h [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+private:
+    DebugDisplayInterface::Color m_textColor;
+    int m_xPos;
+    int m_yPos;
+    int m_width;
+    int m_height;
+    int m_rightMargin;
+    int m_leftMargin;
+};
```
Comment (tomsons26) [inline src/game/client/debugdisplay.h]: missing new line

File: src/game/client/debugdisplay.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            } else {
+                ++v4;
+            }
+        }
+        if (v4 > 0) {
+            Draw_Text(m_xPos + m_rightMargin, m_yPos, v2);
+            m_xPos += v4;
+        }
+    }
+}
```
Comment (tomsons26) [inline src/game/client/debugdisplay.cpp]: missing new line

**Comment by duncanspumpkin** [comment]
Maybe it would be an idea to get clang format going as a ci then it means you don't need to comment on the formatting as much.

File: src/game/client/w3ddebugdisplay.cpp (hunk 1 above)
Comment (OmniBlade) [reply src/game/client/w3ddebugdisplay.cpp]: I believe an equivalent function already exists in colorspace.h but I didn't know the name for it so just made one up.

**Comment by OmniBlade** [comment]
Its on my todo list along with moving to github workflows.

**Pull request merged**